### PR TITLE
[Fix] `Animate 3D graphic` animations

### LINKED ISSUE #344: Add animation to minimap expand/shrink
https://github.com/Esri/arcgis-maps-sdk-swift-samples/blob/05b3a37f1419d9bedc98fe5a82a62635e63a2a4e/Shared/Samples/Animate%203D%20graphic/Animate3DGraphicView.swift#L57-L64

Add animation here so it looks more smooth. Try a shorter than default duration?

```swift
withAnimation {
    isShowingFullMap.toggle()
}
```

### PR DESCRIPTION
**Pull request by CalebRas**
## Description

This PR fixes a bug in the `Animate 3D graphic` sample where the plane animation would slow when a sheet was opened. The fix replaces the `Timer` with a `CADisplayLink` and instead adjusts the speed by skipping frames. As a result, the speed `Slider` in the "Mission Settings"  was replaced with a `Picker`.

This PR also adds an animation to minimap to close #344.

## Linked Issue(s)

- `swift/issues/5130`

## How To Test

- Open the "Mission" sheet in the sample.
- Ensure the animation behind the sheet doesn't slow.
- Use the picker to adjust the speed of the animation.

### CODE REVIEW

File: Shared/Samples/Animate 3D graphic/Animate3DGraphicView.swift [hunk 1]
```diff
@@ -93,15 +96,18 @@ struct Animate3DGraphicView: View {
...
-            model.animation.stop()
-        }
     }
     
     /// The list containing the mission settings.
     private var missionSettings: some View {
         List {
             Section("Mission") {
+                VStack {
+                    StatRow("Progress", value: model.animation.progress.formatted(.rounded))
```
Comment (des12437) [inline Shared/Samples/Animate 3D graphic/Animate3DGraphicView.swift:106]: The `StatRow("Progress")` view could use some top padding.

Mac Catalyst:
Screenshot 2024-03-29 at 1 06 06 PM

**Review by des12437 — DISMISSED** [review]

File: Shared/Samples/Animate 3D graphic/Animate3DGraphicView.Model.swift [hunk 2]
```diff
@@ -223,14 +212,18 @@ extension Animate3DGraphicView {
...
     struct Animation {
-        /// The timer for the animation used to loop through the animation frames.
-        var timer: Timer?
+        /// The timer used to loop through the animation frames.
+        private(set) var displayLink: CADisplayLink?
         
-        /// The speed of the animation used to set the timer's time interval.
-        var speed = 50.0
+        /// The speed of the animation.
+        var speed: AnimationSpeed = .medium
```
Comment (dfeinzimer) [inline Shared/Samples/Animate 3D graphic/Animate3DGraphicView.Model.swift:219]:
```suggestion
        var speed = AnimationSpeed.medium
```

File: Shared/Samples/Animate 3D graphic/Animate3DGraphicView.Model.swift [hunk 3]
```diff
@@ -255,26 +248,31 @@ extension Animate3DGraphicView {
         /// The index of the current frame in the frames list.
         private var currentFrameIndex = 0
         
-        /// Stops the animation by invalidating the timer.
-        mutating func stop() {
-            timer?.invalidate()
-            isPlaying = false
+        /// Sets up the animation using  a given display link.
```
Comment (dfeinzimer) [inline Shared/Samples/Animate 3D graphic/Animate3DGraphicView.Model.swift:251]:
```suggestion
        /// Sets up the animation using a given display link.
```

**Review by dfeinzimer — COMMENTED** [review]
Nice work!

Looks like the speed options are getting cutoff in the list. This seems pre-existing so just noting it here - not related to these changes.

**Comment by CalebRas** [comment]
> Looks like the speed options are getting cutoff in the list. This seems pre-existing so just noting it here - not related to these changes.

@dfeinzimer I'm curious how you are see this. It looks like the "Mission Settings" navigation title isn't showing up for you either. It works fine for me, even on device. What iPhone are you use/OS are you using?

**Comment by dfeinzimer** [comment]
> What iPhone are you use/OS are you using?

2nd gen SE on 17.3.1

**Comment by CalebRas** [comment]
> 2nd gen SE on 17.3.1

@dfeinzimer It looks like this happens when `presentationDetents` and `frame` are used together, which is a common pattern in the samples. I noted it in `swift/issues/5145`, so this PR should be good to go. Thanks for bringing this up!

**Review by dfeinzimer — APPROVED** [review]

**Review by des12437 — APPROVED** [review]

**Comment by CalebRas** [comment]
Thank you for reviewing!

**Pull request merged**